index-configuration-tool/logstash_conf_parser.py

```diff
@@ -0,0 +1,46 @@
+import sys
+from lark import Lark
+from lark import Transformer
+
+
+class LogstashTransformer(Transformer):
+    def var_name(self, s):
+        return s.value
+
+    def string_literal(self, s):
+        (s,) = s
+        return s[1:-1]
```

Comment:
Good point. Parts of this code required some degree of trial-and-error, which is probably why it's not clear. The methods in the Transformer correspond to a `rule_name` or `TERMINAL` in the grammar definition. 

In the case of a `TERMINAL`, the method parameter is a [Token](https://lark-parser.readthedocs.io/en/latest/classes.html#token). In case of a `rule_name`, the method parameter is a `list` of its children in the AST. Note that the Transformer parses the AST bottom-up.

For now, i've added type hints and comments to make this clearer, but i think i can further improve the transformer to make it clearer. I'll work on this after implementing tests.

Reply:
I've cleaned this up now so the Transformer doesn't operate on `TERMINAL` values, and uses only `rule_names`. I've also added comments.